app/classifier/tasks/dropdown/index.cjsx

```diff
@@ -82,23 +82,21 @@ module.exports = React.createClass
         not select.required or (annotation.value[i]?.value? and annotation.value[i]?.value isnt "")
 
   getInitialState: ->
-    selectedOptions: []
+    optionsKeys: {}
 
   componentWillMount: ->
     @menus = []
     @syncAnnotations()
 
-  componentDidMount: ->
-    annotationValues = @props.annotation.value
-    unless annotationValues.length
-      @props.annotation.value = @props.task.selects.map -> {value: null, option: false}
+  componentWillReceiveProps: (nextProps) ->
+    if nextProps.task isnt @props.task
+      @syncAnnotations()
 
+  componentDidMount: ->
     if @props.autoFocus is true
       @handleFocus()
 
   componentDidUpdate: (prevProps) ->
-    if prevProps.task isnt @props.task
-      @syncAnnotations()
```

Comment:
See previous comment. Shouldn't we be waiting for props to update before setting the annotation value.